internal/querycoord/metrics_info.go

```diff
@@ -0,0 +1,84 @@
+// Copyright (C) 2019-2020 Zilliz. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions and limitations under the License.
+
+package querycoord
+
+import (
+	"context"
+
+	"go.uber.org/zap"
+
+	"github.com/milvus-io/milvus/internal/log"
+
+	"github.com/milvus-io/milvus/internal/proto/commonpb"
+	"github.com/milvus-io/milvus/internal/proto/milvuspb"
+	"github.com/milvus-io/milvus/internal/util/metricsinfo"
+	"github.com/milvus-io/milvus/internal/util/typeutil"
+)
+
+// TODO(dragondriver): add more detail metrics
+func getSystemInfoMetrics(
+	ctx context.Context,
+	req *milvuspb.GetMetricsRequest,
+	qc *QueryCoord,
+) (*milvuspb.GetMetricsResponse, error) {
+
+	coordTopology := metricsinfo.CoordTopology{
+		Self: metricsinfo.ComponentInfos{
+			Name: metricsinfo.ConstructComponentName(typeutil.QueryCoordRole, Params.QueryCoordID),
+		},
+		ConnectedNodes: make([]metricsinfo.ComponentInfos, 0),
+	}
+
+	nodesMetrics := qc.cluster.getMetrics(ctx, req)
+	for _, nodeMetrics := range nodesMetrics {
+		if nodeMetrics.err != nil {
+			log.Warn("skip invalid metrics of node",
+				zap.Error(nodeMetrics.err))
+			continue
```

Comment:
If nodeMetrics.err != nil, the error info shouldn't be returned to sdk?

Reply:
Yeah, I think so. Maybe there any exceptions happend on this node, we should filter it.

Reply:
We should focus the whole instead of the part.